common/unsafe/src/main/java/org/apache/spark/unsafe/array/ByteArrayMethods.java

```diff
@@ -47,13 +47,20 @@ public static int roundNumberOfBytesToNearestWord(int numBytes) {
   public static boolean arrayEquals(
       Object leftBase, long leftOffset, Object rightBase, long rightOffset, final long length) {
     int i = 0;
-    while (i <= length - 8) {
-      if (Platform.getLong(leftBase, leftOffset + i) !=
-        Platform.getLong(rightBase, rightOffset + i)) {
-        return false;
-      }
-      i += 8;
-    }
+
+      // This attempts to speed up the memcmp type of operation, but there is no way
+      // to guarantee that the offsets will be on a word boundary in order to use
+      // Platform.getLong
```

Comment:
No.  The only way it can be used with the 8 byte copy is if both leftOffset and rightOffset either start on proper word boundries, or are off of boundries by the same degree.  In practice, this absolute alignment between left and right offsets happens very rarely.  So rarely that the performance impact of checking for it every time negates any advantage you get from performing it.